deploy/vagrant/vbox/buildbox.sh

```diff
@@ -12,9 +14,16 @@ if [ -f tachyon-dev.box ]; then
   rm -f tachyon-dev.box
 fi
 
-vagrant up
-vagrant package --output tachyon-dev.box default
-vagrant destroy -f
+BASE_BOX='chef/centos-7.0'
+AVAIL_BOX=$(vagrant box list | grep $BASE_BOX | grep '(virtualbox, 1.0.0)')
+
+if [ -z "$AVAIL_BOX" ]; then
+  vagrant box add chef/centos-7.0 --provider virtualbox
+fi
+
+echo "Generating 'tachyon-dev.box' based on box '$BASE_BOX' ..."
+vagrant box repackage $BASE_BOX virtualbox 1.0
+mv package.box tachyon-dev.box
```

Comment:
@yuzhu @uronce-cc I saw the ufs-hadoop1 & ufs-hadoop2 download scripts are taking advantage of that. I will do the same with Maven. For the first vagrant run (build our tachyon-dev box), I will map the vagrant sync folder to the parent one and install maven. Then in the second vagrant run (which is the main deploy process), I will be able to bypass the maven installation by looking at the shared folder and /usr/bin/mvn symlink.